### PR TITLE
feat: Create a scheduled job for changing session status on event completion

### LINKED ISSUE #5777: Change the session status from pending to rejected if the event has passed.
**Describe the bug**

Currently if the user's session is in pending state till the event has happened, it still shows pending after the event has passed. A better label will be rejected since the user's session wasn't accepted till the event.

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to 'my sessions after event has passed'
2. Click on 'an older session'
3. See error

**Expected behaviour**

Pending label should be Rejected.

**Additional context**

I am working on this issue.

### PR DESCRIPTION
**Pull request by uds5501**
Fixes #5777

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
It creates a scheduled job for changing session status of a passed event

#### Changes proposed in this pull request:

- Using queries to load up session data specific to requirements
- add a scheduled job regarding the same

### CODE REVIEW

**Comment by uds5501** [comment]
@iamareebjamal @CosmicCoder96 Please have a look :)

File: app/api/helpers/scheduled_jobs.py [hunk 1]
```diff
@@ -46,6 +47,16 @@ def send_after_event_mail():
...
 
 
+def change_session_state_on_event_completion():
+    from app import current_app as app
+    with app.app_context():
+        sessions_to_be_changed = Session.query.join(Event).filter(Session.state == 'pending')\
+                                 .filter(Event.ends_at < datetime.datetime.now())
+        for session in sessions_to_be_changed:
+            session.state = 'rejected'
+            save_to_db(session, 'Changed {} session state to rejected'.format(session.title))
```
Comment (iamareebjamal) [inline app/api/helpers/scheduled_jobs.py]: No need for looping, can be done in a single query

Comment (uds5501) [reply app/api/helpers/scheduled_jobs.py]: @iamareebjamal I tried using a loop directly to test the updates with 

```python
db.session.query(Session).join(Event).filter(Session.state=='confirmed').update({"video_url":'a'})
db.session.commit()
```

and it threw : 

```
sqlalchemy.exc.InvalidRequestError: Can't call Query.update() or Query.delete() when join(), outerjoin(), select_from(), or from_self() has been called

```

How do i tackle this?

Comment (mrsaicharan1) [reply app/api/helpers/scheduled_jobs.py]: you will not be able to use an update alongside a successive join. Why do you need to use `join` here? You can directly use the related `Session`'s event's fields.

Comment (iamareebjamal) [reply app/api/helpers/scheduled_jobs.py]: https://stackoverflow.com/questions/33153823/update-joined-table-via-sqlalchemy-orm-using-session-query

Comment (mrsaicharan1) [reply app/api/helpers/scheduled_jobs.py]: @uds5501 use `db.session.query(Session).filter_by(event_id=your_event_id, state=='confirmed').update(...)`

Comment (uds5501) [reply app/api/helpers/scheduled_jobs.py]: @mrsaicharan1 this query won't check if the event has passed

**Comment by mrsaicharan1** [comment]
Also Check whether the event date is before or equal to the current date
when you're using  filter_by^

**Comment by uds5501** [comment]
@mrsaicharan1 but applying a filter_by in the query you suggested would only use Sessions model, how to access the Event model details?

**Comment by mrsaicharan1** [comment]
Use related event_id

**Pull request merged**